static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -0,0 +1,241 @@
+// @flow
+import fastdom from 'fastdom';
+import { initYoutubePlayer } from 'common/modules/atoms/youtube-player';
+import tracking from 'common/modules/atoms/youtube-tracking';
+import Component from 'common/modules/component';
+import $ from 'lib/$';
+import config from 'lib/config';
+import detect from 'lib/detect';
+import debounce from 'lodash/functions/debounce';
+
+const players = {};
+
+// retrieves actual id of atom without appended index
+const getTrackingId = (atomId: string) => atomId.split('/')[0];
+
+const recordPlayerProgress = (atomId: string) => {
+    const player = players[atomId].player;
+    const pendingTrackingCalls = players[atomId].pendingTrackingCalls;
+
+    if (!pendingTrackingCalls.length) {
+        return;
+    }
+
+    if (!player.duration) {
+        player.duration = player.getDuration();
+    }
+
+    const currentTime = player.getCurrentTime();
+    const percentPlayed = Math.round(currentTime / player.duration * 100);
+
+    if (percentPlayed >= pendingTrackingCalls[0]) {
+        tracking.track(pendingTrackingCalls[0], getTrackingId(atomId));
+        pendingTrackingCalls.shift();
+    }
+};
+
+const killProgressTracker = (atomId: string) => {
+    if (players[atomId].progressTracker) {
+        clearInterval(players[atomId].progressTracker);
+    }
+};
+
+const setProgressTracker = (atomId: string) =>
+    (players[atomId].progressTracker = setInterval(
+        recordPlayerProgress.bind(null, atomId),
+        1000
+    ));
+
+const onPlayerPlaying = atomId => {
+    const player = players[atomId];
+
+    killProgressTracker(atomId);
+    setProgressTracker(atomId);
+    tracking.track('play', getTrackingId(atomId));
+
+    const mainMedia =
+        (player.iframe && player.iframe.closest('.immersive-main-media')) ||
+        null;
+    if (mainMedia) {
+        mainMedia.classList.add('atom-playing');
+    }
+
+    if (
+        player.endSlate &&
+        !player.overlay.parentNode.querySelector('.end-slate-container')
+    ) {
+        player.endSlate.fetch(player.overlay.parentNode, 'html');
+    }
+};
+
+const onPlayerPaused = atomId => killProgressTracker(atomId);
+
+const onPlayerEnded = (atomId: string) => {
+    const player = players[atomId];
+
+    killProgressTracker(atomId);
+    tracking.track('end', getTrackingId(atomId));
+    player.pendingTrackingCalls = [25, 50, 75];
+
+    const mainMedia =
+        (player.iframe && player.iframe.closest('.immersive-main-media')) ||
+        null;
+    if (mainMedia) {
+        mainMedia.classList.remove('atom-playing');
+    }
+};
+
+const STATES = {
+    ENDED: onPlayerEnded,
+    PLAYING: onPlayerPlaying,
+    PAUSED: onPlayerPaused,
+};
+
+const onVideoContainerNavigation = (atomId: string): void => {
```

Comment:
I don't think we have a convention yet, but I prefer prefixing the export to the function expression:

`export const onVideoContainerNavigation`

Means that I have to do less scrolling in the file, and remember less, since when looking at the function I know it is exported... thoughts?